docs/recipes/git-show.rst

```diff
@@ -40,6 +40,45 @@ Show all files in commit
     >>> for e in commit.tree:
     >>>     print(e.name)
 
+======================================================================
+Produce something like a ``git show`` message
+======================================================================
+
+In order to display timezone information you have to create a subclass of
+tzinfo as described in the `Python datetime documentation`_::
+
+    from datetime import tzinfo, timedelta
+    class FixedOffset(tzinfo):
+        """Fixed offset in minutes east from UTC."""
+
+        def __init__(self, offset):
+            self.__offset = timedelta(minutes = offset)
+
+        def utcoffset(self, dt):
+            return self.__offset
+
+        def tzname(self, dt):
+            return None
+
+        def dst(self, dt):
+            return timedelta(0) # dealing with DST would be too complicated
```

Comment:
Many countries do not observe DST and they thus have in effect a fixed-offset timezone (though since they're political the implementation would never be as simple as this). Personally I would not set a timezone and use something like `"{}{}".format(offset / 60, offset % 60)` to print the offset.

Though I wonder if pygit2 should provide a datetime as part of the signature instead of time and offset separately.